### PR TITLE
ENHANCEMENT Support global composer installation

### LINKED ISSUE #21: Installation via composer global
Would be awesome if you could install sspak via composer:

`composer global require "silverstripe/sspak"`

That would install sspak into into `~/.composer/vendor/bin/`, just like how you'd install PHPUnit globally for a user.

### PR DESCRIPTION
**Pull request by robbieaverill**
* Adds a new bin location for composer so we can use `sspak` still as the bin filename
* Rejigged readme to use `#` for titles, and added installation example with composer
* Tweaked a couple of constant values to ensure composer-compatible `sspak` works with B/C over original

Resolves #21

### CODE REVIEW

**Comment by wilr** [comment]
👍

File: bin-composer/sspak [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env php
+<?php
+/**
+ * Run SSPak with composer!
+ */
+define('PACKAGE_ROOT', dirname(__FILE__) . '/../');
+define('SSPAK_COMPOSER_INSTALL', true);
+require_once dirname(__FILE__) . '/../bin/sspak';
```
Comment (dhensby) [inline bin-composer/sspak]: You've just defined `PACKAGE_ROOT` - you should use it

**Comment by dhensby** [comment]
This commit does a lot more than it says on the tin. Would you mind either splitting it into 3 commits or even 3 PRs?

1. Readme format update
2. Code style update
3. Adding composer global support (including new docs)

**Comment by robbieaverill** [comment]
Hey @dhensby - I've pushed an update separating the changes into three commits.

>Is there a reason we can't have a unified bin file/location?

Yes - composer requires a single bin file declares for it to create symlinks. If I tell it to use `bin/sspak` then I don't have the ability to define the package root manually. There's possibly a way around it with some trickery in being able to detect that the package has been installed with composer, but this way allows you to still call `sspak`, but have the logic for it in a separate file.

>You've just defined `PACKAGE_ROOT` - you should use it

Good point - updated.

**Comment by dhensby** [comment]
OK - thanks.

**Pull request merged**